server/specification/specification.go

```diff
@@ -0,0 +1,225 @@
+package specification
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"sync"
+	"text/template"
+
+	"github.com/SKF/go-rest-utility/problems"
+	"go.opencensus.io/trace"
+)
+
+const (
+	swaggerUIDirectory = "swagger-ui"
+	indexFilePath      = "swagger-ui/index.html"
+	openAPIPath        = "/openapi.yaml"
+)
+
+type SwaggerFS struct {
+	FS fs.FS
+
+	SwaggerURLPrefix      string
+	APIEndpointsURLPrefix string
+
+	NotFoundHandler         http.Handler
+	MethodNotAllowedHandler http.Handler
+
+	initOnce      sync.Once
+	compiledFiles map[string]CompiledFile
+}
+
+type swaggerFSHandler struct {
+	*SwaggerFS
+	*Config
+	pathsToCompile []string
+}
+
+type CompiledFile struct {
+	Contents *bytes.Buffer
+	Info     fs.FileInfo
+}
+
+func (swaggerfs *SwaggerFS) Handler(opts ...Option) *swaggerFSHandler {
+
+	config := &Config{
+		swaggerUIDirectory: swaggerUIDirectory,
+		indexFilePath:      indexFilePath,
+		openAPIPath:        openAPIPath,
+	}
+
+	for _, opt := range opts {
+		opt(config)
+	}
+
+	pathsToCompile := []string{
+		config.indexFilePath,
+		config.openAPIPath,
+	}
+
+	if swaggerfs.FS == nil {
+		panic("no swagger filesystem set")
+	}
+
+	if swaggerfs.NotFoundHandler == nil {
```

Comment:
The NotFoundHandler and MethodNotAllowed handlers are optional, so it should be fine if they are nil?

It could even be done in the init function ;)

Reply:
oh, didn't see that they were optional. Will correct that.
Wanted to get failure when creating/assigning the handler so main() fails. Else we have to wait until someone/thing vists docs/swagger*. (We don't usually have test doing that?)

Reply:
All the new services have tests for swagger for this reason. Since it is part of the server I would expect there to be some tests.

Reply:
All the new services have tests for swagger for this reason. Since it is part of the server I would expect there to be some tests.

Reply:
Still waiting for test to fail for something that is known at the start of runtime feels bad.